### PR TITLE
Add LED strip support

### PR DESCRIPTION
**Pull request by catfishjw**
Button control from rov_control should still work. 
To make leds work, the ui needs a dropdown that sends a message asking the node for all programmed animations and adds the returned items to the dropdown. When an animation is selected, send it as a message for the leds to receive.

### CODE REVIEW

File: src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+from common.csm_common_interfaces.msg import PinState
+from std_msgs.msg import String
+import Jetson.GPIO as GPIO
+import rclpy
+from rclpy.node import Node
+from rainbowio import colorwheel
+
+import board
+import neopixel
```
Comment (yameatmeyourdead) [inline src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py:9]: use adafruit-circuitpython-neopixel-spi instead

File: src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+LED_PIN = 0
+BLACK = (0,0,0)
+WHITE = (255,255,255)
+RED = (255, 0, 0)
+YELLOW = (255, 150, 0)
+GREEN = (0, 255, 0)
+CYAN = (0, 255, 255)
+BLUE = (0, 0, 255)
+PURPLE = (180, 0, 255)
+PIXELS = neopixel.NeoPixel(LED_PIN, NUM_LIGHTS)
```
Comment (yameatmeyourdead) [inline src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py:23]: pixels = neopixel.NeoPixel_SPI(board.SPI(), NUM_PIXELS)

**Review by yameatmeyourdead — CHANGES_REQUESTED** [review]
Use SPI version of the library instead

File: src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py (hunk 1 above)
Comment (catfishjw) [reply src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py]: done

File: src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                    PIXELS.brightness(0)
+                lastAnimation = msg.data
+            elif msg.data in self.colors:
+                lastColor = ledColor
+                if ledState and (lastColor != msg.data):
+                    exec(f"self.{lastAnimation}({msg.data}, 0.1)")()
+
+
+    def on_set_rov_gpio(self, message: PinState):
+        if message.pin not in LED_PIN:
```
Comment (yameatmeyourdead) [inline src/rov/LEDController/src/rov/LEDCOntroller/LEDController.py:87]: bruh

**Review by yameatmeyourdead — CHANGES_REQUESTED** [review]

**Review by yameatmeyourdead — APPROVED** [review]
litmus

**Pull request merged**